### PR TITLE
Add netstandard build

### PR DESCRIPTION
**Pull request by ah-**
I'd love to use GitVersion with .net core.
GitVersion on .net core means it'll be easy to integrate it with the dotnet cli tooling.
Plus it will also possible to compile it to a native binary and use it in any Linux/Mac project, no matter if that project is .NET or not.

This is a first go at getting GitTools.Core building. I've implemented it similarly to how the current NET40/NET45 builds already work.

Could you have a look?

### CODE REVIEW

File: src/GitTools.Core/GitTools.Core.NETSTANDARD/project.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+﻿{
+  "supports": {},
+  "dependencies": {
+    "LibGit2Sharp-dotnet": "1.0.0",
+    "LibGit2Sharp.NativeBinaries-dotnet": "[1.0.137]",
```
Comment (ah-) [inline src/GitTools.Core/GitTools.Core.NETSTANDARD/project.json]: Currently uses my fork of LibGit2Sharp with .net core support. Will update this to the official packages once they are available.

File: src/GitTools.Core/GitTools.Core.Shared/Exceptions/WarningException.cs [hunk 2]
```diff
@@ -1,19 +1,25 @@
 namespace GitTools
 {
     using System;
+#if !NETSTANDARD
     using System.Runtime.Serialization;
+#endif
```
Comment (ah-) [inline src/GitTools.Core/GitTools.Core.Shared/Exceptions/WarningException.cs]: Binary serialization has been deprecated in .net core. Most projects handle it by either completely removing [Serializable] or by using ifdefs, I've gone with the latter so nothing changes for classic .net.

File: src/GitTools.Core/GitTools.Core.Shared/Git/Helpers/GitRepositoryHelper.cs [hunk 3]
```diff
@@ -12,7 +12,7 @@
     /// </summary>
     public static class GitRepositoryHelper
     {
-        static readonly ILog Log = LogProvider.GetCurrentClassLogger();
+        static readonly ILog Log = LogProvider.GetLogger(typeof(GitRepositoryHelper));
```
Comment (ah-) [inline src/GitTools.Core/GitTools.Core.Shared/Git/Helpers/GitRepositoryHelper.cs]: GetCurrentClassLogger isn't available in portable LibLog, I think this is equivalent.

File: src/GitTools.Core/GitTools.Core.Shared/Helpers/ProcessHelper.cs [hunk 4]
```diff
@@ -53,23 +57,37 @@ public static int Run(Action<string> output, Action<string> errorOutput, TextRea
             if (output == null)
                 throw new ArgumentNullException("output");
 
-            workingDirectory = workingDirectory ?? Environment.CurrentDirectory;
+            workingDirectory = workingDirectory ?? Directory.GetCurrentDirectory();
```
Comment (ah-) [inline src/GitTools.Core/GitTools.Core.Shared/Helpers/ProcessHelper.cs]: Another deprecated API with a simple replacement.

**Comment by asbjornu** [comment]
@ah- Thanks for this, but as you can see, Travis and AppVeyor are both failing. Think you can address this in any way? We would like to keep our builds green. 😄

**Comment by ah-** [comment]
Yep, will investigate.

File: src/GitTools.Core/GitTools.Core.NETSTANDARD/project.json (hunk 1 above)
Comment (JakeGinnivan) [inline src/GitTools.Core/GitTools.Core.NETSTANDARD/project.json]: Hey, I don't want to fork the Ecosystem, this has downstream implications so we can't merge this until these libraries natively add netstandard support. Sorry

**Pull request closed without merging**